src/main.rs

```diff
@@ -2519,6 +2519,27 @@ fn run_steps(ib: &mut ImageBuilder) -> Result<()> {
                     &a.publisher,
                 ])?;
             }
+            "pkg_set_p5p_publisher" => {
+                #[derive(Deserialize)]
+                struct PkgSetP5PPublisherArgs {
+                    archive: String,
+                    publisher: String,
+                }
+
+                let a: PkgSetP5PPublisherArgs = step.args()?;
+                let mp = ib.root()?;
+
+                pkg(log, &["-R", &mp.to_str().unwrap(),
+                    "set-publisher", "-p", &a.archive, "--search-first",
+                ])?;
```

Comment:
This form of `set-publisher` does a bit of magic when you don't specify a publisher argument, as far as I can tell.  Depending on what publishers are in the archive it might end up adding more than one, or updating existing publishers.  Is that what you intend?

Also, it seems like the image will contain publisher information that might not make sense once it's installed, unless you ship the archive at the exact same location inside the final image.

Did you try a version of this that used the temporary archive functionality; i.e., something like `pkg update -g $ARCHIVE_PATH` ?  You would still need to have set the existing publisher(s) non-sticky I believe.